core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/validation/ValidateRequest.java

```diff
@@ -102,13 +107,17 @@ private ValidationResult doValidate(BranchContext context, Writer index) throws
 		final List<Promise<Object>> validationPromises = Lists.newArrayList();
 		// evaluate selected rules
 		for (ValidationRule rule : rules) {
+			RemoteJob remoteJob = context.service(RemoteJob.class);
 			checkArgument(rule.getCheckType() != null, "CheckType is missing for rule " + rule.getId());
 			final ValidationRuleEvaluator evaluator = ValidationRuleEvaluator.Registry.get(rule.getType());
 			if (evaluator != null) {
 				validationPromises.add(pool.submit(rule.getCheckType(), () -> {
```

Comment:
It would be great to reduce pressure on the validation pool by not submitting more and more rules to execution once we know that the operation was canceled by the user. Please add a check before this line.